chsdi/views/features.py

```diff
@@ -429,7 +429,11 @@ def _get_features_for_filters(params, layerBodIds, maxFeatures=None, where=None)
             # Filter by time instant
             if params.timeInstant is not None and hasattr(model, '__timeInstant__'):
                 timeInstantColumn = model.time_instant_column()
-                query = query.filter(timeInstantColumn == params.timeInstant)
+                timeInstantIndex = layerBodIds.index(layer)
+                timeInstant = params.timeInstant[timeInstantIndex]
```

Comment:
yes you're right, not 100% safe. I would do that earilier during the validation process.

```
len(layers) == len(timestamps)
````
and if we have only one timestamp apply it to all the time enabled layers.

It's also an undocumented parameter, so we may be safe. thanks for your comment anyway, I'll adapt now.